packages/data-service/src/data-service.ts

```diff
@@ -300,37 +332,120 @@ class DataService extends EventEmitter {
           // @ts-expect-error Callback without result...
           return callback(this._translateMessage(error));
         }
-        (callback as Callback<CollectionInfo[]>)(null, data!);
+        callback?.(
+          null,
+          (data ?? []).map((coll) =>
+            adaptCollectionInfo({ db: databaseName, ...coll })
+          )
+        );
       }
     );
   }
 
+  /**
+   * Special list collections method that handles pulling collection names from
+   * user privileges when user is not allowed to listCollections on a database.
+   * Compared to the listCollections, this method will never throw
+   *
+   * TODO: Only needed while we are waiting for authorizedCollections: true
+   * support to land in the nodejs driver
+   */
+  listCollectionsNamesOnly(
+    dbName: string,
+    callback: Callback<ReturnType<typeof adaptCollectionInfo>[]>
+  ): void {
+    const listCollectionsAsync = promisify(this.listCollections.bind(this)) as (
+      // Promisify doesn't extract correct types from the method because of the
+      // overloads so we have to convince it that the arguments we provide are
+      // fine
+      dbName: string,
+      filter: Document,
+      options: { nameOnly: true; authorizedCollections?: true }
+    ) => Promise<ReturnType<typeof adaptCollectionInfo>[]>;
+
+    void Promise.all([
```

Comment:
we already started using `async / await` do we need to keep adding callback style methods here? won't we just have to convert them later?

Reply:
Ahh, sorry, at a quick glance all of them looked callback-styled so I was preserving the interfaces more or less. If it's fine to start adding async methods, I'll totally will change this 👍

Reply:
Done in d67891d75cfb28248fd4e821971d89c1dbe07902